### ISSUE TITLE
Kubernetes Dashboard not loading after upgrading to Kubernetes version 1.16.7

### ISSUE THREAD

**Issue by raswinraaj** [Azure/AKS#1600]
**What happened**:
I upgraded my AKS cluster from 1.15.7 to 1.16.7 from the portal's "Upgrade" section. After the upgrade happened successfully i tried to browse the kubernetes dashboard page using the below command 
`az aks browse --resource-group aks-dev-rg --name aks-dev
`
Kubernetes dashboard https://127.0.0.1:8001/ didnot respond .
![dashboard](https://user-images.githubusercontent.com/50330414/81527783-cd6c3200-9378-11ea-94ab-5e8dda5980f6.png)

**What you expected to happen**:
Kubernetes dashboard page should load
**How to reproduce it (as minimally and precisely as possible)**:
Upgrade cluster from 1.15.7 to 1.16.7

**Anything else we need to know?**:
I also tried to delete the clusterrolebinding for kubernetes dashboard and re-creating it. That didnt work as well.
```
kubectl delete ClusterRoleBinding kubernetes-dashboard
kubectl create clusterrolebinding kubernetes-dashboard --clusterrole=cluster-admin --serviceaccount=kube-system:kubernetes-dashboard
```
**Environment**:
- Kubernetes version (use `kubectl version`):
`Client Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.0", GitCommit:"e8462b5b5dc2584fdcd18e6bcfe9f1e4d970a529", GitTreeState:"clean", BuildDate:"2019-06-19T16:40:16Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"windows/amd64"}
Server Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.7", GitCommit:"e2d9f8479783020904aba3de7499a49be6c75ebd", GitTreeState:"clean", BuildDate:"2020-04-09T02:31:15Z", GoVersion:"go1.13.6", Compiler:"gc", Platform:"linux/amd64"}`
- Size of cluster (how many worker nodes are in the cluster?):2
- General description of workloads in the cluster (e.g. HTTP microservices, Java app, Ruby on Rails, machine learning, etc.): dotnet core apps. All applications are accessible and running without any issue
- Others:

**Comment by jellebens** [comment]
![image](https://user-images.githubusercontent.com/1615249/81540481-03f27e80-9372-11ea-9fdb-8ae742fdd271.png)

I get this error

**Comment by Kanth81** [comment]
I received the following error.
This page isn’t working localhost is currently unable to handle this request.
HTTP ERROR 502.
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.0"
Server Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.7"

**Comment by 4c74356b41** [comment]
this is what I'm getting:
```
Valid from December 31, 0000 to December 31, 0000
Serial Number: 509ad9c67a960665769d6f0cfb2bbc0a
Signature Algorithm: ecdsa-with-SHA256
```

I dont think I can do anything about it. pretty much nothing works with this certificate...

**Comment by majorimi** [comment]
Any update on this? I'm still getting this error....

**Comment by george-zenfolio** [comment]
I got this to work for me....
Get the token for the dashboard out of this secret, remember to base64 decode it. 
`kubectl get ServiceAccount kubernetes-dashboard -n kube-system -o json | jq '.secrets | .[] | .name`
Re-add the admin role, but use different name for the binding. 
`kubectl create clusterrolebinding kubernetes-dashboard-admin --clusterrole=cluster-admin --serviceaccount=kube-system:kubernetes-dashboard`

Then my dashboards loaded.

**Comment by GaurangRLB** [comment]
Got the same issue after updating to 1.16.9. Any update on this?

**Comment by 4c74356b41** [comment]
this seems to be fixed with 1.18.x, but its in preview. the workaround would be to delete the kubedashboard addon from AKS and install it manually or something. MS doesnt seem to case about this issue somehow. even when talking to PG they just ignore you.

**Comment by raswinraaj** [comment]
The workaround I got was to use --admin at the end when logging in using az
aks command. It looks like the dashboard can be viewed only as an admin.

**Comment by 4c74356b41** [comment]
it doesnt matter, thats a different "issue" (not an issue tbh). so `az aks browse` works somehow, but `kubectl port-forward` won't help, as certificate is not accepted by any sane browser.

**Comment by markmcgookin** [comment]
I am having this issue as well.... what's weird is, we have two clusters, both upgraded from 1.14.x to 1.15.x then to 1.16.9 today one worked fine and dashboard worked fine (although you need to pick a kubeconfig file now). I usually use the azure cloud shell for this stuff, but you need to do the get-credentials locally, then you can still use that file.

The dashboard doesn't seem to be working for our other cluster though, which uses identical configuration... but if I get-credentials with -a at the end locally, and then do the browse command locally via azure cli it let me select the new config file and I was able to connect.

It might be caching the kubeconfig file I uploaded to connect to the first cluster, which wouldn't have had permissions in for the second cluster as I hadn't merged them in yet. Either way, get-credentials with -a created a kubeconfig file that let me connect. Thanks to @raswinraaj for the advice.

**Comment by robbiezhang** [comment]
AKS upgraded the kube-dashboard to v2 (from v1) for 1.16 cluster due to the incompatible k8s API change. However, we find several issues with the v2 dashboard
1. HTTPS is enforced in the v2 dashboard
The old CLI implementation based on kubectl port-forward no longer works for HTTPS. We have updated the CLI and CLI extension to use kube-apiserver proxy in March 2020. Customer need to latest azure-cli. If you install the azure-cli extension for AKS, they need to run `az extension update --name aks-preview` to get it up-to-date.
2. The new sign-in process for the v2 dashboard.
The v2 dashboard uses a different sign-in approach. You need to follow the new process to get access to the dashboard.
https://docs.microsoft.com/en-us/azure/aks/kubernetes-dashboard#sign-in-to-the-dashboard-kubernetes-116

3. Refresh the kubeconfig.
Notice that it's better to get the cluster credential again if they use the user token in the kubeconfig to sign-in, since we renew the user token every 6 months automatically. The old kubeconfig might still work to access the cluster because it contains a valid certificate, but the user token might already invalid.

4. 401 unauthorized error due to Chrome update.
We use kube-dashboard v2.0.0-r3 for 1.16 cluster, because this is the last version which is fully compatible with 1.16 cluster. However, it doesn't include a security fix to set the secure site cookie from the client. It causes the issue in the recent incidents, since chrome started to enforce the restriction in July update and rolled out to 100% in August. https://www.chromium.org/updates/same-site 
The security fix is included in v2.0.0-r4, however, the version is marked as "might not be compatible with 1.16".

So, there are 3 options for mitigation:
1. Use Firefox to access the dashboard. Verified that the FF hasn't enforce the restriction. However, we are not sure when FF will adopt the restriction, and causes the break again.
2. Upgrade the cluster to 1.17, and the dashboard will be upgraded to 2.0.0-rc7 automatically.
3. Deploy the kube-dashboard on your own, prefer in a different namespace to avoid conflict.
